File: pinyougou_manager/pinyougou_manager_interface/src/main/java/com/pinyougou/manager/service/BrandService.java
package com.pinyougou.manager.service;

import com.pinyougou.common.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    /**
     * 好习惯：在接口上加注释
     * 品牌列表
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 分页查询
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 新增品牌
     * @param brand
     * @return
     */
    void add(TbBrand brand);

    /**
     * 根据品牌ID查询品牌对象
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 修改
     * @param brand
     */
    public void update(TbBrand brand);

    /**
     * 根据多个ID来删除品牌
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 根据对象分页搜索
     * @param pageNum
     * @param pageSize
     * @param brand
     * @return
     */
    PageResult search(Integer pageNum, Integer pageSize, TbBrand brand);

    /**
     * 获取Map数据
     * @return
     */
    List<Map> findMap();
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService",function($http){
    this.findAll=function(){
        return $http.get("../brand/findAll");
    }
    this.findPage=function(pageNum,pageSize){
        return $http.get("../brand/findPage?pageNum="+pageNum+"&pageSize="+pageSize);
    }
    this.update=function(entity){
        return $http.post("../brand/update",entity);
    }
    this.add=function(entity){
        return $http.post("../brand/add",entity);
    }
    this.dele=function(ids){
        return $http.get("../brand/delete?ids="+ids);
    }
    this.findOne=function (id) {
        return $http.get("../brand/findOne?id="+id);
    }
    this.search=function(pageNum,pageSize,entity){
        return $http.post("../brand/search?pageNum="+pageNum+"&pageSize="+pageSize,entity);
    }
    this.findMap=function(){
        return $http.get("../brand/findMap");
    }
});